### ISSUE TITLE
[icetray] I3::dataio::open doesn't work with latest libarchive (Trac #2398)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#2390]
I3::dataio::open tries a number of methods to decompress files 
on cvmfs py3-v4.0.1 with libarchive 3.3.2 this works fine. However, on py3-4.1.0 has libarchive 3.4.0 the stream returns gibberish. This prevents clsim from working because clsim uses I3::dataio::open to read its safe prime file. A trivial example is given below:

```text
#include <iostream>                                                                                                  
#include <icetray/open.h>

int main(){
  std::string safeprimes_file("/cvmfs/icecube.opensciencegrid.org/data/safeprimes_base32.gz");

  boost::iostreams::filtering_istream ifs;
  I3::dataio::open(ifs, safeprimes_file);
  char tag[18];
  ifs.read(tag, 17);
  std::cout << tag << '\\' << std::endl;                                          
  std::cout << "EOF: " << ifs.eof() << std::endl;                                                       
}              

```

Migrated from https://code.icecube.wisc.edu/ticket/2398
```json
{
    "status": "closed", 
    "changetime": "2020-06-24T12:31:42", 
    "description": "I3::dataio::open tries a number of methods to decompress files \non cvmfs py3-v4.0.1 with libarchive 3.3.2 this works fine. However, on py3-4.1.0 has libarchive 3.4.0 the stream returns gibberish. This prevents clsim from working because clsim uses I3::dataio::open to read its safe prime file. A trivial example is given below:\n\n\n{{{\n#include <iostream>                                                                                                  \n#include <icetray/open.h>\n\nint main(){\n  std::string safeprimes_file(\"/cvmfs/icecube.opensciencegrid.org/data/safeprimes_base32.gz\");\n\n  boost::iostreams::filtering_istream ifs;\n  I3::dataio::open(ifs, safeprimes_file);\n  char tag[18];\n  ifs.read(tag, 17);\n  std::cout << tag << '\\\\' << std::endl;                                          \n  std::cout << \"EOF: \" << ifs.eof() << std::endl;                                                       \n}              \n\n}}}\n", 
    "reporter": "kjmeagher", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1593001902142004", 
    "component": "combo core", 
    "summary": "[icetray] I3::dataio::open doesn't work with latest libarchive", 
    "priority": "blocker", 
    "keywords": "clsim libarchive", 
    "time": "2020-01-09T16:17:13", 
    "milestone": "Autumnal Equinox 2020", 
    "owner": "olivas", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20200109T20:41:31: 
* olivas changed status from "new" to "accepted"
* olivas changed owner from "" to "olivas"

**Comment by icecube-trac** [comment]
Trac update at 20200116T20:02:42: david.schultz commented:

> I've investigated, and something in libarchive changed.  We have this comment in our code:
> 
> 
> ```text
> /*
>  * NB: libarchive exposes raw data as an archive with a
>  * single entry whose pathname is "data".
>  */
> ```
> 
> which is apparently no longer true, as it now reports "safeprimes_base32".
> 
> It appears that it was this commit: https://github.com/libarchive/libarchive/commit/0debf359fc2d4796aedab04d96c6e2046551b56e
> which has the message "support reading metadata from compressed files".
> 
> It looks like when certain conditions are met, it will return the actual filename now.
> 
> Will continue to look into this and hopefully have a patch.

**Comment by icecube-trac** [comment]
Trac update at 20200116T23:32:10: 
* david.schultz commented:

> In changeset 178501:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="178501"
> put a shim in archive_filter to handle new metadata provided by libarchive 3.4, specifically the archive name. this fixes #2390. note that auto-detection of archive type (tar vs raw) is inherently fragile and not recommended by libarchive devs.
> ```


* david.schultz changed status from "accepted" to "closed"
* david.schultz changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20200624T12:31:42: 
* olivas changed milestone from "Vernal Equinox 2020" to "Autumnal Equinox 2020"
* olivas commented:

> Milestone Vernal Equinox 2020 deleted
